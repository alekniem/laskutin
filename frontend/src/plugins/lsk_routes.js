import LskPageIndexBiller from "@/components/page/LskPageIndexBiller";
import LskPageIndexCustomer from "@/components/page/LskPageIndexCustomer";
import LskPageIndexInvoice from "@/components/page/LskPageIndexInvoice";

import LskPageShowBiller from "@/components/page/LskPageShowBiller";
import LskPageShowCustomer from "@/components/page/LskPageShowCustomer";
import LskPageShowInvoice from "@/components/page/LskPageShowInvoice";

import LskPageLogin from "@/components/page/LskPageLogin";
import LskPageNotFound from "@/components/page/LskPageNotFound";

export default [
    {
        path: "/",
        name: "index",
    },
    {
        path: "/login",
        name: "login",
        component: LskPageLogin,
    },
    {
        path: "/biller",
        name: "indexBiller",
        component: LskPageIndexBiller,
    },
    {
        path: "/biller/:id",
        name: "showBiller",
        component: LskPageShowBiller,
        props: true,
    },
    {
        path: "/customer",
        name: "indexCustomer",
        component: LskPageIndexCustomer,
    },
    {
        path: "/customer/:id",
        name: "showCustomer",
        component: LskPageShowCustomer,
        props: true,
    },
    {
        path: "/invoice",
        name: "indexInvoice",
        component: LskPageIndexInvoice,
    },
    {
        path: "/invoice/:id",
        name: "showInvoice",
        component: LskPageShowInvoice,
        props: true,
    },
    {
        path: "*",
        component: LskPageNotFound,
    },
];
