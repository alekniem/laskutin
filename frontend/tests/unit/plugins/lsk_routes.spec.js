import lsk_routes from "@/plugins/lsk_routes";

describe("lsk_routes", () => {
    it("should match snapshot", () => {
        expect(Array.isArray(lsk_routes)).toBe(true);
        const snapshot = [];
        lsk_routes.forEach((routeItem) => {
            expect(Object.prototype.toString.call(routeItem)).toBe("[object Object]");
            const item = {};
            Object.keys(routeItem).forEach((key) => {
                if (key === "component") {
                    if (Object.prototype.hasOwnProperty.call(routeItem[key], "name")) {
                        item[key] = { name: routeItem[key]["name"] };
                    }
                } else {
                    item[key] = routeItem[key];
                }
            });
            snapshot.push(item);
        });
        expect(snapshot).toMatchSnapshot();
    });
});
