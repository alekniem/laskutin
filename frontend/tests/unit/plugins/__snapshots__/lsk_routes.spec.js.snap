// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lsk_routes should match snapshot 1`] = `
Array [
  Object {
    "name": "index",
    "path": "/",
  },
  Object {
    "component": Object {
      "name": "LskPageLogin",
    },
    "name": "login",
    "path": "/login",
  },
  Object {
    "component": Object {
      "name": "LskPageIndexBiller",
    },
    "name": "indexBiller",
    "path": "/biller",
  },
  Object {
    "component": Object {
      "name": "LskPageShowBiller",
    },
    "name": "showBiller",
    "path": "/biller/:id",
    "props": true,
  },
  Object {
    "component": Object {
      "name": "LskPageIndexCustomer",
    },
    "name": "indexCustomer",
    "path": "/customer",
  },
  Object {
    "component": Object {
      "name": "LskPageShowCustomer",
    },
    "name": "showCustomer",
    "path": "/customer/:id",
    "props": true,
  },
  Object {
    "component": Object {
      "name": "LskPageIndexInvoice",
    },
    "name": "indexInvoice",
    "path": "/invoice",
  },
  Object {
    "component": Object {
      "name": "LskPageShowInvoice",
    },
    "name": "showInvoice",
    "path": "/invoice/:id",
    "props": true,
  },
  Object {
    "component": Object {
      "name": "LskPageNotFound",
    },
    "path": "*",
  },
]
`;
